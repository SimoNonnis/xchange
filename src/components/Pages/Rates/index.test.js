import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Rates } from './index';

const pocketsInfo = {
  GBP: {
    code: 'GBP',
    name: 'British Pound',
    symbol: '£'
  },
  EUR: {
    code: 'EUR',
    name: 'Euro',
    symbol: '€'
  },
  USD: {
    code: 'USD',
    name: 'American Dollar',
    symbol: '$'
  },
  RUB: {
    code: 'RUB',
    name: 'Russian Ruble',
    symbol: '₽'
  }
};

const rates = {
  GBP: 1000,
  EUR: 2000,
  USD: 3000
};

describe('Test <Rates />', () => {
  it('should render with required props', () => {
    const wrapper = shallow(
      <Rates
        currenciesList={['GBP', 'EUR', 'USD']}
        pocketsInfo={pocketsInfo}
        rates={rates}
        selected="RUB"
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
