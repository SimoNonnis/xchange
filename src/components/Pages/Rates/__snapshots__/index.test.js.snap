// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Rates /> should render with required props 1`] = `
<div>
  <nav
    aria-label="Main navigation"
    className="navigation"
    role="navigation"
  >
    <Link
      replace={false}
      to="/exchange"
    >
      <ReactComponent
        className="icon"
      />
    </Link>
  </nav>
  <h1
    className="pageTitle"
  >
    Rates Page
  </h1>
  <div
    className="rates-container"
  >
    <Rate
      amount={1000}
      key="GBP"
      name="British Pound"
      selectedCode="RUB"
    />
    <Rate
      amount={2000}
      key="EUR"
      name="Euro"
      selectedCode="RUB"
    />
    <Rate
      amount={3000}
      key="USD"
      name="American Dollar"
      selectedCode="RUB"
    />
  </div>
</div>
`;
